firebase-firestore/src/test/java/com/google/firebase/firestore/core/QueryTest.java

```diff
@@ -615,6 +615,94 @@ public void testImplicitOrderBy() {
         baseQuery.orderBy(orderBy("foo", "desc")).orderBy(orderBy("bar", "asc")).getOrderBy());
   }
 
+  @Test
+  public void testImplicitOrderByInMultipleInequality() {
```

Comment:
Can we also add the following tests?

1. A mix of explicit order by and implicit order by fields from MIEQ
2. Exercise the normalization logic that uses the last explicit order by direction, i.e., `SELECT * FROM k WHERE a > 1 ORDER BY b DESC` normalizes to `SELECT * FROM k WHERE a > 1 ORDER BY b DESC, a DESC, __key__ DESC`